__tests__/app.test.js

```diff
@@ -26,3 +26,44 @@ describe('GET /api', () => {
       });
   });
 });
+
+describe('GET /api/topics', () => {
+  test('200: Responds with an array of topics, each with a slug and description', () => {
+    return request(app)
+      .get('/api/topics')
+      .expect(200)
+      .then(({ body: { result } }) => {
+        expect(Array.isArray(result)).toBe(true);
+        expect(result).toHaveLength(3);
+        result.forEach((result) => {
+          expect(result).toEqual(
+            expect.objectContaining({
+              slug: expect.any(String),
+              description: expect.any(String),
+            })
+          );
+        });
+      });
+  });
+
+  test('404: Responds with an error message when the endpoint does not exist', () => {
+    return request(app)
+      .get('/api/nonexistent-endpoint')
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe('Not found');
+      });
+  });
+
+  test('500: Responds with an error message for a server error', () => {
+    jest.spyOn(db, 'query').mockImplementationOnce(() => {
+      throw new Error('Database error');
+    });
+    return request(app)
+      .get('/api/topics')
+      .expect(500)
+      .then(({ body }) => {
+        expect(body.msg).toBe('Internal Server Error');
+      });
+  });
+});
```

Comment:
This is a really interesting way to test for the 500 error, but generally we never want to see this error as its a fallback incase something goes wrong outside of what we generally expect, but its a nice test to see here none the less.